src/NumberTheory/QuadFormAndIsom/finite_group_actions.jl

```diff
@@ -0,0 +1,871 @@
+      special::Bool=false,
+      check::Bool=true,
+      kwargs...,
+    ) -> MatrixGroup
+
+Return the pointwize stabilizer in the orthogonal group of ``L`` of the lattice
+``S``, contained in the rational span of ``L``.
+
+The implementation requires that the largest saturated submodule ``K`` of ``L``
+orthogonal to ``S`` is definite.
+
+If `stable` is set to `true`, return the largest subgroup of the stable
+orthogonal group $O^\#(L)$ fixing pointwize ``S``. If `special` is set
+to `true`, return the largest subgroup of the special orthogonal group
+$SO(L)$ fixing pointwize ``S``. If both keyword arguments are `true`,
+then return the intersection of the corresponding pointwize stabilizers.
+
+If `check` is set to true, the function tests whether ``S`` is contained in
+$L\otimes \mathbb{Q}$ and whether the lattice ``K`` is definite.
+
+The function first computes the orthogonal group of ``K``: the extra keyword
+arguments in `kwargs` are optional arguments in the computations of such a
+group (see [`isometry_group(::ZZLat)`](@ref)).
+
+# Examples
+```jldoctest
+julia> A4 = root_lattice(:A, 4)
+Integer lattice of rank 4 and degree 4
+with gram matrix
+[ 2   -1    0    0]
+[-1    2   -1    0]
+[ 0   -1    2   -1]
+[ 0    0   -1    2]
+
+julia> A2 = lattice_in_same_ambient_space(A4, QQ[1 0 0 0; 0 1 0 0])
+Integer lattice of rank 2 and degree 4
+with gram matrix
+[ 2   -1]
+[-1    2]
+
+julia> H = pointwize_stabilizer_in_orthogonal_group(A4, A2)
+Matrix group of degree 4
+  over rational field
+
+julia> order(H)
+2
+```
+"""
+function pointwize_stabilizer_in_orthogonal_group(
+    L::ZZLat,
+    S::ZZLat;
+    kwargs...
+  )
+  @req ambient_space(L) === ambient_space(S) "Lattices are not contained in the same quadratic space"
+  return stabilizer_in_orthogonal_group(L, basis_matrix(S); kwargs...)
+end
+
+@doc raw"""
+    _stabilizer_in_diagonal_action(
+      L::ZZLat,
+      K::ZZLat,
+      N::ZZLat,
+      OK::MatrixGroup,
+      ON::MatrixGroup,
+    ) -> Vector{QQMatrix}
+
+Given a primitive extension ``K\oplus N \subseteq L`` of even lattices,
+and given two finite groups of isometries ``OK`` and ``ON`` of ``K`` and ``N``,
+respectively, return generators for the stabilizer of ``L`` in ``OK\times ON``,
+seen as a group of isometries of the ambient space of ``L``.
+"""
+function _stabilizer_in_diagonal_action(
+    L::ZZLat,
+    K::ZZLat,
+    N::ZZLat,
+    OK::MatrixGroup,
+    ON::MatrixGroup;
+    gen_fun::Function=gens,
+  )
+  # Necessary tests and can speed up kernel computations
+  @assert is_finite(OK)
+  @assert is_finite(ON)
+
+  # Need the glue map to determine isometries of OK\times ON preserving L in
+  # K^\vee\oplus N^\vee
+  phi, HKinqK, HNinqN = glue_map(L, K, N)
+  iphi = inv(phi)
+
+  gen = QQMatrix[]
+  qN = codomain(HNinqN)
+  genN = ZZMatrix[matrix(hom(qN, qN, elem_type(qN)[qN(lift(a)*matrix(g)) for a in gens(qN)])) for g in gen_fun(ON)]
+  GN = Oscar._orthogonal_group(qN, genN; check=false)
```

Comment:
How about something like
```suggestion
  discN = discriminant_representation(N, ON; ambient_representation=true, check=false)
  GN = discN(ON)
```